flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/JdbcDialect.java

```diff
@@ -31,8 +31,13 @@
 
 import static java.lang.String.format;
 
-/** Handle the SQL dialect of jdbc driver. */
-@Internal
+/**
+ * Represents a dialect of SQL implemented by a particular JDBC system. Dialects should be immutable
+ * and stateless.
+ *
+ * @see JdbcDialectFactory
+ */
+@PublicEvolving
 public interface JdbcDialect extends Serializable {
```

Comment:
Since we're making this a public API now it would be good to go through all the JavaDocs in here and bring them a bit into shape.